src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -24,109 +24,193 @@
 #include <variant>
 
 #include "util/Algorithm.h"
+#include "util/ComparisonWithNan.h"
 #include "util/ConfigManager/ConfigExceptions.h"
 #include "util/ConfigManager/ConfigOption.h"
 #include "util/ConfigManager/ConfigShorthandVisitor.h"
 #include "util/ConfigManager/ConfigUtil.h"
 #include "util/ConfigManager/generated/ConfigShorthandLexer.h"
 #include "util/ConfigManager/generated/ConfigShorthandParser.h"
 #include "util/Exception.h"
+#include "util/Forward.h"
 #include "util/StringUtils.h"
 #include "util/TypeTraits.h"
 #include "util/antlr/ANTLRErrorHandling.h"
 #include "util/json.h"
 
 namespace ad_utility {
 // ____________________________________________________________________________
-void ConfigManager::verifyHashMapEntry(
-    std::string_view jsonPathToEntry,
-    const HashMapEntry::pointer entryPointer) {
+ConfigManager::HashMapEntry::HashMapEntry(Data&& data)
+    : data_{std::make_unique<Data>(std::move(data))} {}
+
+// ____________________________________________________________________________
+bool ConfigManager::HashMapEntry::holdsConfigOption() const {
+  // Make sure, that it is not a null pointer.
+  AD_CORRECTNESS_CHECK(data_);
+
+  return std::holds_alternative<ConfigOption>(*data_);
+}
+
+// ____________________________________________________________________________
+bool ConfigManager::HashMapEntry::holdsSubManager() const {
   // Make sure, that it is not a null pointer.
-  AD_CORRECTNESS_CHECK(entryPointer != nullptr);
+  AD_CORRECTNESS_CHECK(data_);
+
+  return std::holds_alternative<ConfigManager>(*data_);
+}
+
+// ____________________________________________________________________________
+std::optional<ConfigOption*> ConfigManager::HashMapEntry::getConfigOption()
+    const {
+  if (holdsConfigOption()) {
+    return std::get_if<ConfigOption>(data_.get());
+  } else {
+    return std::nullopt;
+  }
```

Comment:
Factor out those two repetitive patterns (they are the same for `ConfigManager` and `ConfigOption` (ohter than the types . E.G. factor out a template with the common implementation.

Reply:
Done, see [954abdf](https://github.com/ad-freiburg/qlever/pull/1153/commits/954abdfc48b6e5751bffee2ae9b37df1538f0cbf).